pallets/phala/src/lib.rs

```diff
@@ -628,90 +628,80 @@ impl<T: Trait> Module<T> {
 		Ok(())
 	}
 
-	/// Try to remove a registered worker from the registry by its `machine_id`
-	///
-	/// Take the storage items (WorkerState and MachineOwner) if they present. Otherwise returns
-	/// None. It doesn't change the stash info. It doesn't emmit any events, since it's up to the
-	/// caller to decide if the WorkerState is updated or not.
-	fn take_machine_if_present(machine_id: &Vec<u8>)
-	-> Option<(T::AccountId, WorkerInfo<T::BlockNumber>)> {
-		if !MachineOwner::<T>::contains_key(machine_id) {
-			return None;
-		}
-		let stash = MachineOwner::<T>::take(machine_id);
-		let worker_info = WorkerState::<T>::take(&stash);
-		Some((stash, worker_info))
+	/// Unlinks a worker from a stash account. Only call when they are linked.
```

Comment:
What if a unlinked worker call this? should we `ensure!` a worker linked a statsh first?

Reply:
ok I found it only be called in `register_worker_internal`